R/agyw-integration.R

```diff
@@ -0,0 +1,1517 @@
+                          summarise_age_cat_male(df1, "Y025_049"),
+                          summarise_age_cat_male(df1, "Y015_049"))
+
+  # Calculate incidence
+  df3 <- dplyr::bind_rows(df1, df2) %>%
+    dplyr::mutate(incidence_cat = cut(incidence,
+                                      c(0, 0.3, 1, 3, 10^6),
+                                      labels = c("Low", "Moderate", "High", "Very High"),
+                                      include.lowest = TRUE, right = TRUE))
+
+
+
+  #' Check that sum of disaggregated infections is the same as total infections
+  #  TO DO: add warning for sum not matching - contact admin
+  sum_infections <- df3$infections_nosex12m + df3$infections_sexcohab + df3$infections_sexnonreg + df3$infections_msm + df3$infections_pwid
+
+  stopifnot(max(df3$infections - sum_infections) < 10^{-9})
+
+
+  df3 %>%
+    dplyr::mutate(concat = paste0(area_id, age_group), iso3 = options$area_scope) %>%
+    dplyr::select(area_id, age_group, concat,
+                  nosex12m, sexcohab, sexnonregplus, sexnonreg, msm, pwid,
+                  iso3, area_level,
+                  population, plhiv, infections, incidence, incidence_cat,
+                  prev_nosex12m, prev_sexcohab,
+                  prev_sexnonreg, prev_msm, prev_pwid,
+                  rr_msm, rr_pwid, rr_sexnonreg,
+                  population_nosex12m, population_sexcohab,
+                  population_sexnonreg, population_msm, population_pwid,
+                  plhiv_nosex12m, plhiv_sexcohab,
+                  plhiv_sexnonreg, plhiv_msm, plhiv_pwid,
+                  susceptible_nosex12m, susceptible_sexcohab,
+                  susceptible_sexnonreg, susceptible_msm, susceptible_pwid,
+                  incidence_nosex12m, incidence_sexcohab,
+                  incidence_sexnonreg, incidence_msm, incidence_pwid,
+                  infections_nosex12m,infections_sexcohab,
+                  infections_sexnonreg, infections_msm, infections_pwid) %>%
+    dplyr::mutate_if(is.numeric, as.numeric) %>%
+    dplyr::mutate_if(is.factor, as.character)
+
+
+
+}
+
+
+#' Calculate incidence in high risk male key populations
+#'
+#' @param outputs Naomi output.
+#' @param options Naomi options extracted from outputs
+#' @param male_srb Estimates of male sexual risk groups generated by `agyw_adjust_sexbehav_msm_pwid()`
+#' @param male_logit_prevalence Risk adjusted estimates of male prevalence in sexual risk groups generated by `agyw_calculate_prevalence_male()`
+#' @param survey_year NOTES:: add in description + when this should be adjusted. Hardcoded to 2018.
+#'
+#' @return Wide format output required for the AGYW workbook
+#'
+#' @export
+#'
+#' Survey year should be updated to most current household survey in the country -
+#' for countries without recent household surveys, leave at 2018 - the spatiotemporal
+#' model of sexual behaviour fitted to all countries has the most data for in roughly 2018
```

Comment:
Can you fix the docs here, couple of things
* Update the parameters
* The details about "survey year" should either be under the parameter, after the title or under a `@details` section. This won't go into the generated docs where it is atm
* Does this need to be exported? I think it is only used internally in which case replace the `export` with `@keywords internal`

Reply:
Thanks - have updated.